src/Cryptography/Sodium/Bindings/Signing.hs

```diff
@@ -0,0 +1,328 @@
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_sign_seed_keypair"
+  cryptoSignSeedKeyPair ::
+    -- | Pointer to the @public key@. It has a length of 'cryptoSignPublicKeyBytes'.
+    Ptr CUChar ->
+    -- | Pointer to the @secret key@. It has a length of 'cryptoSignSecretKeyBytes'.
+    Ptr CUChar ->
+    -- | Pointer to the @seed@. It has a length of 'cryptoSignSeedBytes'.
+    Ptr CUChar ->
+    -- | Return code is 0 on success, -1 on error.
+    IO CInt
+
+-------------------
+-- Combined Mode --
+-------------------
+
+-- | The 'cryptoSign' function prepends a signature to a message @m@, whose length is @mlen@ bytes,
+-- using the secret key @secret key@.
+--
+-- The signed message, which includes the signature plus an unaltered copy of the message, is put
+-- into @signed message@ and is 'cryptoSignBytes' + @mlen@ bytes long.
+--
+-- If @smlen@ is not a 'Foreign.nullPtr', then the actual length of the signed message is stored in
+-- @smlen@.
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_sign"
+  cryptoSign ::
+    -- | Pointer to the @signed message@.
+    Ptr CUChar ->
+    -- | Pointer to the length of the @signed message@.
+    Ptr CULLong ->
+    -- | Pointer to the @message@ to sign.
+    Ptr CUChar ->
+    -- | Length of the @message@.
+    CULLong ->
+    -- | Pointer to the @secret key@.
+    Ptr CUChar ->
+    -- | Return code is 0 on success, -1 on error.
+    IO CInt
+
+-- | The 'cryptoSignOpen' function checks that the signed message @signed message@,
+-- whose length is @smlen@ bytes, has a valid signature for the public key @public key@.
+--
+-- On success, it puts the message without the signature into @m@, stores its length in @mlen@
+-- if @mlen@ is not a 'Foreign.nullPtr' pointer.
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_sign_open"
+  cryptoSignOpen ::
+    -- | Pointer to the @message@ you wish to sign.
+    Ptr CUChar ->
+    -- | Pointer to the length of the @message@.
+    Ptr CULLong ->
+    -- | Pointer to the @signed message@.
+    Ptr CUChar ->
+    -- | Length of the signed message.
+    CULLong ->
+    -- | Pointer to the @public key@.
+    Ptr CUChar ->
+    -- | On success, the function returns 0
+    -- If the signature isn't valid, then the function returns -1.
+    IO CInt
+
+-------------------
+-- Detached Mode --
+-------------------
+
+-- | The 'cryptoSignDetached' function signs the message @m@, whose length is @mlen@ bytes,
+-- using the secret key @secret key@ and puts the signature into @sig@, which can be up to
+-- 'cryptoSignBytes' bytes long.
+-- The actual length of the signature is put into @siglen@ if @siglen@ is not 'Foreign.nullPtr'.
+-- It is safe to ignore @siglen@ and always consider a signature as 'cryptoSignBytes' bytes long;
+-- shorter signatures will be transparently padded with zeros if necessary.
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_sign_detached"
+  cryptoSignDetached ::
+    -- | Pointer to the @signature@.
```

Comment:
Why is 'signature' monospaced?

Reply:
I thought it would emphasise the subject? But it can be removed. :)

Reply:
I think it's better: being monospaced like that suggests a named parameter or a code snippet.